reactioncommerce/reaction
# Fixes #4037: Paginate and Filter Orders Subscription on Admin Dashboard

Resolves #4037 
Impact: **critical**  
Type: **bugfix|performance**

## Issue
The `CustomPaginatedOrders` subscription doesn't paginate the query and pulls all the orders from the DB. When you have an Orders collection of over 700 records this resulted in an interrupted query and on the Orders Dashboard panel never loading the orders.

It throws the following error:
```
Exception in added observe/observeChanges callback: MongoError: Sort exceeded memory limit of 104857600 bytes, but did not opt in to external sorting. Aborting operation. Pass allowDiskUse:true to opt in. 
```

## Solution
Paginate and query on the `PaginatedOrders` subscription method. 

## Steps to Reproduce
Context: Logged in as Admin
1. Add over 700 orders to the database
2. Try to open the Orders menu from the Admin panel
3. See the eternal loading icon and the stack trace on the server


## Breaking changes
Shouldn't have any. Added two optional parameters to one function

## Testing
1. Add over 700 records to the Orders collection.
2. Go to the Admin dashboard and open the Orders menu
3. You should see all the orders and filter them without any problem.
